pkg/generate/app/componentref.go

```diff
@@ -299,6 +302,9 @@ type ComponentInput struct {
 
 	ExpectToBuild bool
 	ScratchImage  bool
+	// For keeping track of if this image was retrieved from a Dockerfile,
+	// which changes our BuildConfig results
```

Comment:
> it's kinda scotch-tape

:+1: 

> but couldn't find a better way to do it. I'm definitely open to suggestions!

I think the approach you described is perfectly valid, no concerns there.